pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -69,6 +69,9 @@ type ResolvedPipelineRunTask struct {
 	ResolvedTaskResources *resources.ResolvedTaskResources
 	// ConditionChecks ~~TaskRuns but for evaling conditions
 	ResolvedConditionChecks TaskConditionCheckState // Could also be a TaskRun or maybe just a Pod?
+	// this flag is set for a task when task result resolution fails for that task
+	// and mainly used to create a task run object with failure for that task
+	InvalidTaskResultsInFinally bool
```

Comment:
wondering if there's a way to avoid saving execution state in resolvedpipelineruntask because it no longer means that it's a PipelineTask with its references resolved (looking into refactoring so that we don't put the taskrun in this struct as well)....we'd initially considered saving the skipped state here when working on when expressions, but decided with calling the method twice (when skipping and when updating status)

cc @bobcatfish

Reply:
PipelineTask being skipped due to when expression failure is a different scenario then final task skip/failing due to task result resolution failure.

When expression is part of `ResolvedPipelineRunTask` and can be evaluated from `GetPipelineConditionStatus` i.e. we can call `Skip` again and will receive accurate result (once a PipelineTask is declared skipped, `Skip` would return true for that PipelineTask based on WhenExpression evaluation inside of `Skip`).

With final task being skipped/failed due to task results resolution failure is determined while retrieving execution queue and not scheduled if task result resolution fails. There is no means of storing such information in `ResolvedPipelineRunTask` except this flag `InvalidTaskResultsInFinally`. 

The key difference is a source of truth available in `ResolvedPipelineRunTask` to all the reconciling cycles v/s reconciler making decision in one cycle and making it available all subsequent cycles.

I hope this make sense 😜 

I will be happy to hear if we can still avoid such flags (not in favor of it either).

Reply:
Yes, that makes sense

Maybe in a future PR after we do some refactoring, we can avoid having that flag there